pkg/alertmanager/multitenant.go

```diff
@@ -1002,6 +1040,27 @@ func (am *MultitenantAlertmanager) serveRequest(w http.ResponseWriter, req *http
 
 	if ok {
 		userAM.mux.ServeHTTP(w, req)
+
+		// If needed, update the last time the Alertmanager received requests.
+		if _, ok := am.receivingRequests.Load(userID); ok {
+			level.Debug(am.logger).Log("msg", "updating last request reception time", "user", userID)
+			am.receivingRequests.Store(userID, time.Now().Unix())
+		}
+		return
+	}
+
+	// If the Alertmanager initialization was skipped, start the Alertmanager.
+	if _, ok := am.receivingRequests.Load(userID); ok {
+		userAM, err = am.startAlertmanager(req.Context(), userID)
```

Comment:
This seems like a more interesting metric to collect: When a skipped config was initialized because of a request. This will give us a feel for flappiness and how effectively our idle timeout value is.

Reply:
A per-tenant metric would be kind of high in cardinality, and we can assess the flappiness of a given tenant by using something like `group by (pod) (cortex_alertmanager_config_hash{user="<user>"}`